src/large_pages/node_large_page.cc

```diff
@@ -339,20 +339,23 @@ __attribute__((__noinline__))
 MoveTextRegionToLargePages(const text_region& r) {
   void* nmem = nullptr;
   void* tmem = nullptr;
-  int ret = 0;
-
-  size_t size = r.to - r.from;
   void* start = r.from;
+  size_t size = r.to - r.from;
+
+  auto free_mems = OnScopeLeave([nmem, tmem, size]() {
```

Comment:
```suggestion
  auto free_mems = OnScopeLeave([&nmem, &tmem, size]() {
```
Otherwise you're copying them by value when the closure is created (i.e., here) and they're still `nullptr` at that point.

Reply:
Of course 🤦 Thanks!